src/Runners/PHPUnit/Options.php

```diff
@@ -327,4 +331,35 @@ private function isFile(string $file): bool
     {
         return file_exists($file) && !is_dir($file);
     }
+
+    /**
+     * Return number of CPU cores, use 2 as fallback.
+     *
+     * Used to set number of processes if argument is set to "auto", allows for portable defaults for doc and scripting.
+     */
+    private function getNumberOfCPUCores(): int
+    {
+        $cores = 2;
+        if (is_file('/proc/cpuinfo')) {
+            // Linux (and potentially Windows with linux sub systems)
+            $cpuinfo = file_get_contents('/proc/cpuinfo');
+            preg_match_all('/^processor/m', $cpuinfo, $matches);
+            $cores = count($matches[0]);
+        } elseif (DIRECTORY_SEPARATOR === '\\') {
+            // Windows
+            if (($process = @popen('wmic cpu get NumberOfCores', 'rb')) !== false) {
+                fgets($process);
+                $cores = (int) fgets($process);
+                pclose($process);
+            }
+        } elseif (($process = @popen('sysctl -a', 'rb')) !== false) {
```

Comment:
you can just query the number by using `sysctl -n hw.ncpu`. No need to parse the output